### PR TITLE
#163002323 Refactor the server files

### PR DESCRIPTION
**Pull request by theghostyced**
#### What does this PR do?
Refactors the server files 
#### Description of Task to be completed?
- Clean up helpers folder
- Clean up the routes folder
- Write tests for the notification system
- Document changes in README.md file
#### How should this be manually tested?
Check the server folders
#### Any background context you want to provide?
Nil
#### What are the relevant pivotal tracker stories?
[#163002323](https://www.pivotaltracker.com/story/show/163002323)
#### Screenshots (if appropriate)
Nil
#### Questions:

### CODE REVIEW

File: server/helpers/article/articleHelper.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    });
+
+    return article;
+  }
+
+  /**
+   * @description Calculates the time it takes to read an article
+   * @param {string} article
+   * @returns {Integer} Time tkaen to read article
```
Comment (Danielshow) [inline server/helpers/article/articleHelper.js]: There is a typo here

File: server/helpers/sendgrid/emailTemplate.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const emailTemplate = {
         `
     },
     welcome: {
-        from: 'no-reply@authorhaven.com',
+        from: {email: 'no-reply@authorhaven.com'},
         subject: 'Authors Haven accout created',
```
Comment (Danielshow) [inline server/helpers/sendgrid/emailTemplate.js]: @theghostyced Please, help fix this typo

File: server/helpers/sendgrid/emailTemplate.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ const emailTemplate = {
          `
     },
     confirmation: {
-        from: 'no-reply@authorhaven.com',
+        from: {email: 'no-reply@authorhaven.com'},
         subject: 'Account succesfully confirmed',
```
Comment (Danielshow) [inline server/helpers/sendgrid/emailTemplate.js]: There is a typo here

File: server/helpers/validation/validations.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+/* eslint-disable max-len */
+/**
+ * @class
+ */
+class Validation {
```
Comment (Danielshow) [inline server/helpers/validation/validations.js:5]: Love this👍

File: server/middlewares/validations/emailValidation.js [hunk 5]
```diff
@@ -69,6 +69,27 @@ class EmailValidations {
...
+   */
+  static doesResetPasswordEmailExist(req, res, next) {
+    User.findOne({
+      where: {
+        email: req.body.email.trim()
+        }
+      }).then(data => {
+        if (!data || data === null) {
+          return res.status(400).json({
+            success: 'failed',
```
Comment (Danielshow) [inline server/middlewares/validations/emailValidation.js]: `success: false` is better

File: tests/middlewares/roleAuthorization.test.js [hunk 6]
```diff
@@ -97,5 +121,11 @@ describe('#Role Middleware', () => {
       expect(response).to.be.a('boolean');
       expect(response).to.equal(true);
     });
+
+    it('should return fale if user is not authorized', () => {
```
Comment (Danielshow) [inline tests/middlewares/roleAuthorization.test.js]: There is a typo here

**Review by Danielshow — COMMENTED** [review]
Nice one @theghostyced - Just implement the little changes

File: server/helpers/article/articleHelper.js (hunk 1 above)
Comment (theghostyced) [reply server/helpers/article/articleHelper.js]: Resolved @Danielshow

**Review by Danielshow — APPROVED** [review]
LGTM

File: server/middlewares/validations/emailValidation.js (hunk 5 above)
Comment (theghostyced) [reply server/middlewares/validations/emailValidation.js]: Resolved @Danielshow

**Review by efalayi — APPROVED** [review]
LGTM

**Pull request merged**